src/alan_opcode_reference.md

```diff
@@ -0,0 +1,860 @@
+## Alan Opcode Reference
+
+The [Alan Standard Library](./standard_library.md), [AMM](./amm_language_reference.md), and [AGA](./aga_language_reference.md) formats all make extensive use of opcodes defined within the runtime to do useful work.
+
+The Alan runtime opcodes are identified with a 64-bit integer and have exactly 3 64-bit arguments. The 64-bit integers have been chosen to match a space-padded string of the name of the opcode (therefore allowing the opcodes to have names from 1-8 ASCII characters long). Most opcodes use the first two arguments as input addresses and the last argument as an output address, but some treat them as constants, event ids, or ignored values.
+
+The following table will specify the opcode name followed by the three arguments (arg0, arg1, arg2) with the kind of argument given (IN = Input Address, OUT = Output Address, CONST = Constant Value, EVENT = Event Name, NULL = Ignored) and then a short description of what that opcode does.
```

Comment:
let's briefly mention what the UNSAFE qualifier

Reply:
Yes, that makes sense